### PR TITLE
docs: Update Machine ID video and placement

### PR DESCRIPTION
**Pull request by alexfornuto**
In #23394 I erroneously uncommented a video, not realizing that the linked video was not relevant to the topic in the page. This PR corrects.

### CODE REVIEW

**Review by zmb3 — APPROVED** [review]

**Comment by zmb3** [comment]
You could instead consider one of these videos:
- https://www.youtube.com/watch?v=LHJf-dbxwpM
- https://www.youtube.com/watch?v=uJvpupYeTS4&t=1523s
- https://www.youtube.com/watch?v=QWd0eqIa9mA

**Review by strideynet — APPROVED** [review]

**Comment by alexfornuto** [comment]
> 

The last link seems like a good candidate. What do you think, @strideynet?

**Comment by alexfornuto** [comment]
I went with the third video @zmb3 suggested, and adjusted it's placement in the guide since it's an overview and not a demo as the existing copy suggested. Re-requested review from @zmb3 since the PR is now different.

**Review by zmb3 — APPROVED** [review]

File: docs/pages/machine-id/introduction.mdx [hunk 1]
```diff
@@ -8,6 +8,16 @@ certificates to any service account (e.g., a CI/CD server) by retrieving
 credentials from the Teleport Auth Service. This enables fine-grained
 role-based access controls and audit.
 
+<iframe
```
Comment (zmb3) [inline docs/pages/machine-id/introduction.mdx:11]: Is this actually how we embed videos now?

Comment (alexfornuto) [reply docs/pages/machine-id/introduction.mdx:11]: It looks like we use different methods for YouTube vs local files. This is the method I found already in use for YT.

**Pull request merged**